src/test/regress/expected/multi_select_distinct.out

```diff
@@ -245,6 +245,882 @@ EXPLAIN (COSTS FALSE)
+         ->  HashAggregate
+               Group Key: remote_scan.worker_column_2
+               ->  Custom Scan (Citus Real-Time)
+                     Task Count: 4
+                     Tasks Shown: One of 4
+                     ->  Task
+                           Node: host=localhost port=57637 dbname=regression
+                           ->  HashAggregate
+                                 Group Key: l_suppkey
+                                 ->  Seq Scan on lineitem_hash_part_360038 lineitem_hash_part
+(14 rows)
+
+-- check the plan if the hash aggreate is disabled
+SET enable_hashagg TO off;
+EXPLAIN (COSTS FALSE)
+	SELECT DISTINCT ceil(count(case when l_partkey > 100000 THEN 1 ELSE 0 END) / 2) AS count 
+		FROM lineitem_hash_part 
+		GROUP BY l_suppkey
+		ORDER BY 1;
+                                                                                                                                                                               QUERY PLAN                                                                                                                                                                                
+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+ Sort
+   Sort Key: ceil(((COALESCE((pg_catalog.sum((ceil(((COALESCE((pg_catalog.sum((ceil(((COALESCE((pg_catalog.sum((ceil(((COALESCE((pg_catalog.sum(remote_scan.count))::bigint, '0'::bigint) / 2))::double precision))))::bigint, '0'::bigint) / 2))::double precision))))::bigint, '0'::bigint) / 2))::double precision))))::bigint, '0'::bigint) / 2))::double precision)
+   ->  Unique
+         ->  Sort
+               Sort Key: ceil(((COALESCE((pg_catalog.sum((ceil(((COALESCE((pg_catalog.sum(remote_scan.count))::bigint, '0'::bigint) / 2))::double precision))))::bigint, '0'::bigint) / 2))::double precision)
+               ->  GroupAggregate
+                     Group Key: remote_scan.worker_column_2
+                     ->  Sort
+                           Sort Key: remote_scan.worker_column_2
+                           ->  Custom Scan (Citus Real-Time)
+                                 Task Count: 4
+                                 Tasks Shown: One of 4
+                                 ->  Task
+                                       Node: host=localhost port=57637 dbname=regression
+                                       ->  HashAggregate
+                                             Group Key: l_suppkey
+                                             ->  Seq Scan on lineitem_hash_part_360038 lineitem_hash_part
+(17 rows)
+
+		
+SET enable_hashagg TO on;
+-- distinct on array_agg results
+SELECT DISTINCT array_agg(l_linenumber) 
```

Comment:
Could we maybe order by the length of the array so that we can easily look for duplicates? And maybe `LIMIT 15` or so?